src/core/public/keyboard_shortcut/keyboard_shortcut_service.ts

```
@@ -40,37 +36,15 @@ export class KeyboardShortcutService {
     this.namespacedIdToKeyLookup.clear();
   }
 
-  private getNormalizedKey = (str: string): string => str.toLowerCase();
-
   private getNamespacedId = (shortcut: Pick<ShortcutDefinition, 'id' | 'pluginId'>) =>
     `${shortcut.id.toLowerCase()}.${shortcut.pluginId.toLowerCase()}`;
 
-  private getEventKeyString = (event: KeyboardEvent): string => {
-    let key = '';
-
-    if (event.ctrlKey) {
-      key += 'ctrl+';
-    }
-
-    if (event.altKey) {
-      key += 'alt+';
-    }
-
-    if (event.shiftKey) {
-      key += 'shift+';
-    }
-
-    if (event.metaKey) {
-      key += 'cmd+';
+  private register(shortcut: ShortcutDefinition): void {
+    if (!this.keyParser.isValidKeyString(shortcut.keys)) {
+      return;
```

lets throw an error here to let devs know they screwed up